api/api.go

```diff
@@ -741,11 +741,8 @@ func (c *Client) websocket(endpoint string, q *QueryOptions) (*websocket.Conn, *
 
 	// check resp status code, as it's more informative than handshake error we get from ws library
 	if resp != nil && resp.StatusCode != 101 {
-		var buf bytes.Buffer
-		io.Copy(&buf, resp.Body)
 		resp.Body.Close()
-
-		return nil, nil, fmt.Errorf("Unexpected response code: %d (%s)", resp.StatusCode, buf.Bytes())
+		return nil, nil, fmt.Errorf("Unexpected response code: %d", resp.StatusCode)
```

Comment:
The status code alone isn't very helpful. Is the response body gzipped? Could we recover the plaintext of the error message somehow?

If it's considerable effort this is ok, but I'm concerned it will leave users with no avenue forward for diagnosing what went wrong.

Reply:
Ah woops good call, can definitely check for gzip and un-gzip, so the error would end up looking like

 ```
"Unexpected response code: 301 (<a href=\"/v1/client/allocation/exec?co
mmand=%5B%22command%22%5D&amp;task=bar&amp;tty=false\">Moved Permanently</a>.\n\n)"```